napari/components/layerlist.py

```diff
@@ -179,3 +185,74 @@ def toggle_selected_visibility(self):
         for layer in self:
             if layer.selected:
                 layer.visible = not layer.visible
+
+    def save(
+        self,
+        path: str,
+        *,
+        selected: bool = False,
+        plugin: Optional[str] = None,
+    ):
+        """Save all or only selected layers to a path using writer plugins.
+
+        If ``plugin`` is not provided and only one layer is targeted, then
+        we just directly call ``napari_write_<layer_type>`` hook specification
+        (see :ref:`single layer writer hookspecs
+        <write-single-layer-hookspecs>`) which will loop through
+        implementations and stop when the first one returns a non-``None``
+        result. The order in which implementations are called can be changed
```

Comment:
It is wrong - needs to be fixed

Reply:
actually. it's correct, (in a `firstresult` hook, it goes until it gets a `not None` result... logic is [here](https://github.com/napari/napari/blob/e78647206109d915d9a9baa788a1e31f1e53d6eb/napari/plugins/_hook_callers.py#L214-L217)).  which means we need to modify our implementations slightly (this was an oversight in the first place):  As a general pattern, hooks should return `None` if they "abstain" or do nothing.  So I'll make all the `napari_write_layer` specs return `None` or a string (the path that was written), rather than a `bool`.